### PR TITLE
Update Tests

### PR DESCRIPTION
**Pull request by AndrewReitz**
* Updated Tests for 3.0.0 of Android Plugin. Dropped support for
all older version of the AGP.
* Fixed minor issue with annotation processing and skipJavaC

### CODE REVIEW

File: groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/FullCompilationSpec.groovy [hunk 1]
```diff
@@ -126,14 +126,9 @@ class FullCompilationSpec extends AndroidFunctionalSpec {
...
     javaVersion                     | _androidPluginVersion | gradleVersion
-    'JavaVersion.VERSION_1_7'       | '2.2.0'              | '3.0'
-    'JavaVersion.VERSION_1_7'       | '2.2.0'              | '3.1'
-    'JavaVersion.VERSION_1_6'       | '2.2.3'              | '3.2'
-    'JavaVersion.VERSION_1_7'       | '2.3.0'              | '3.3'
-    'JavaVersion.VERSION_1_7'       | '2.3.0'              | '3.4'
-    'JavaVersion.VERSION_1_7'       | '2.3.1'              | '3.5'
-    'JavaVersion.VERSION_1_7'       | '2.3.2'              | '3.5'
-    'JavaVersion.VERSION_1_7'       | '2.3.3'              | '4.2'
+    'JavaVersion.VERSION_1_7'       | '3.0.0'              | '4.2'
```
Comment (friederbluemle) [inline groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/FullCompilationSpec.groovy]: You might want to also add Gradle version `4.1` here, which is still the default for a new project created with Android Studio 3.0

File: groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/KotlinSupportSpec.groovy [hunk 2]
```diff
@@ -17,17 +17,20 @@ package groovyx.functional
 
 import groovyx.functional.internal.AndroidFunctionalSpec
 
+import java.lang.Void as Should
```
Comment (friederbluemle) [inline groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/KotlinSupportSpec.groovy]: Are there any non-standard characters on this line? Or is it just the way GitHub renders spaces in import statements? I am seeing two red squares...

screen shot 2017-11-17 at 11 37 11 am

File: groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/FullCompilationSpec.groovy (hunk 1 above)
Comment (AndrewReitz) [reply groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/FullCompilationSpec.groovy]: Thanks for pointing that out. I'll add 4.1 as well.

File: groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/KotlinSupportSpec.groovy (hunk 2 above)
Comment (AndrewReitz) [reply groovy-android-gradle-plugin/src/test/groovy/groovyx/functional/KotlinSupportSpec.groovy]: Those are just spaces. GitHub for some reason does that when using import aliasing.

**Pull request merged**